lib/services/watcher_test.go

```diff
@@ -1095,7 +1096,12 @@ func TestSAMLIdPServiceProviderWatcher(t *testing.T) {
 	// The first event is always the current list of service providers.
 	select {
 	case changeset := <-w.Collector.Cfg.SAMLIdPServiceProvidersC:
-		require.Empty(t, cmp.Diff(types.SAMLIdPServiceProviders{sp1}, changeset,
+		expected := types.SAMLIdPServiceProviders{sp1}
+		sortedChangeset := changeset
```

Comment:
I don't think this assignment does anything, you are effectively sorting `changeset` when you sort `sortedChangeset`.

Same for others.